File: core/src/main/java/com/mcreater/amclcore/model/oauth/MinecraftProductRequestModel.java
package com.mcreater.amclcore.model.oauth;

import com.mcreater.amclcore.annotations.OAuthLoginModel;
import com.mcreater.amclcore.annotations.RequestModel;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
@OAuthLoginModel
@RequestModel
public class MinecraftProductRequestModel {
    private List<MinecraftProductItemModel> items;
    private String signature;
    private String keyId;

    @Data
    @Builder
    @OAuthLoginModel
    @RequestModel
    public static class MinecraftProductItemModel {
        private String name;
        private String signature;
    }
}

File: core/src/main/java/com/mcreater/amclcore/model/oauth/MinecraftProductRequestModel.kt
package com.mcreater.amclcore.model.oauth

import com.mcreater.amclcore.annotations.OAuthLoginModel
import com.mcreater.amclcore.annotations.RequestModel

@RequestModel
@OAuthLoginModel
data class MinecraftProductRequestModel(
    var items: List<MinecraftProductItemModel?>? = null,
    var signature: String? = null,
    var keyId: String? = null
) {
    @RequestModel
    @OAuthLoginModel
    data class MinecraftProductItemModel(
        var name: String? = null,
        var signature: String? = null
    )
}
